packages/components/src/__future__/Tag/Tag.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { ReactNode, HTMLAttributes } from "react"
+import classNames from "classnames"
+import { Icon } from "@kaizen/component-library/components/Icon"
+import tagIcon from "@kaizen/component-library/icons/tag.icon.svg"
+import { OverrideClassName } from "~types/OverrideClassName"
+
+/** @todo Add list of all icons names */
+export type TagIcons = "tag"
+
+export type TagColors =
+  | "gray"
+  | "blue"
+  | "green"
+  | "yellow"
+  | "orange"
+  | "red"
+  | "purple"
+
+export interface TagProps
+  extends OverrideClassName<HTMLAttributes<HTMLSpanElement>> {
+  children: ReactNode
+  color?: TagColors
+  icon?: TagIcons
```

Comment:
We've settled on a `JSXElement` for the `Icons`. For example, consumers will be passing `<AddIcon />` to this prop.